frontend/packages/ceph-storage-plugin/src/components/kms-config/utils.tsx

```diff
@@ -95,22 +90,63 @@ export const createKmsResources = (kms: KMSConfig, update = false, previousData?
     VAULT_CLIENT_CERT: kms.clientCert?.metadata.name,
     VAULT_CLIENT_KEY: kms.clientKey?.metadata.name,
     VAULT_NAMESPACE: kms.providerNamespace,
+    VAULT_TOKEN_NAME: kms.token.value ? KMSSecretName : '',
+    VAULT_CACERT_FILE: kms.caCertFile,
+    VAULT_CLIENT_CERT_FILE: kms.clientCertFile,
+    VAULT_CLIENT_KEY_FILE: kms.clientKeyFile,
   };
 
-  const configMapObj: ConfigMapKind = {
+  const csiConfigObj: ConfigMapKind = {
     apiVersion: ConfigMapModel.apiVersion,
     kind: ConfigMapModel.kind,
     data: {
-      ...configData,
+      [`${kms.name.value}`]: JSON.stringify(csiConfigData),
     },
     metadata: {
-      name: KMSConfigMapName,
+      name: KMSConfigMapCSIName,
       namespace: CEPH_STORAGE_NAMESPACE,
     },
   };
 
-  const csiConfigData: KMSConfigMap = {
-    KMS_PROVIDER: 'vaulttokens',
+  const csiKmsResources: Promise<K8sResourceKind>[] =
+    /** kms.token.value === "" if we are calling this function from StorageClass KMS flow.
+     * Hence, we need to createCommonKmsResources, else it is already created if calling from OCS wizard flow.
+     */
+    kms.token.value ? [] : createCommonKmsResources(kms);
```

Comment:
```suggestion
  
```

Reply:
why you still need to check token when you are in CSI specific function

Reply:
const csiKmsResources: Promise<K8sResourceKind>[] = [createCommonKmsResources(kms)]

Reply:
because I am calling this CSI specific function from createClusterKmsResources (cluster wizard flow) function as well...
so, if I am calling it from createClusterKmsResources function, then I do not have to call createCommonKmsResources function again, as it was already called once in createClusterKmsResources function.

Reply:
Ah! ok understood